crates/pm_auth/src/connector/plaid/transformers.rs

```diff
@@ -42,6 +44,26 @@ impl TryFrom<&types::LinkTokenRouterData> for PlaidLinkTokenRequest {
                     },
                 )?,
             },
+            android_package_name: match item.request.client_platform {
+                api_models::enums::ClientPlatform::Android => {
+                    Some(item.request.android_package_name.clone().ok_or(
+                        errors::ConnectorError::MissingRequiredField {
+                            field_name: "android_package_name",
+                        },
+                    )?)
```

Comment:
why are you raising an error if `None` but wrapping it in `Some` again? If this is a required_field then in the struct type itself register `android_package_name` as `String` instead of `Option<String>`. Or is it required only in certain cases?

Reply:
Since we match on client_platform (which is sent by SDK) and only then take a decision. It's only required in very specific flows.